qir/stdlib/lib/stdlib-cxx.ll

```diff
@@ -0,0 +1,621 @@
+; ModuleID = 'src/stdlib.cpp'
+source_filename = "src/stdlib.cpp"
```

Comment:
I think we should be checking in the `bc` files instead of `ll`. /CC @swernli @bettinaheim

Reply:
I'm torn on this one, because I do want to use IR as the source of truth if possible, but that means we end up checking in (in either format) the `target datalayout` and `target triple` of whoever last generated the file. I can't use this file directly in Linux because the target triple/datalayout is for Mac, so I have to delete those lines or rerun the compilation. Compiling again not only updates the file with my local datalayout/triple but also makes a few other minor changes to the rest of the file (differences in C++ -> IR with AppleClang vs Clang-11). I'm worried checking in IR in any form other than hand-written is going to cause headaches in git and confusion in expected usage.

Reply:
@swernli  IIRC you have to manually set a datalayout during `llvm-as` to force a layout that isn't in the IR.

Reply:
If a file doesn't have any data layout string then it just uses the LLVM default. From the [docs](https://llvm.org/docs/LangRef.html#langref-datalayout):

> When constructing the data layout for a given target, LLVM starts with a default set of specifications which are then (possibly) overridden by the specifications in the datalayout keyword.

But what I've observed is that since QAT combines the stdlib IR and the program IR, LLVM C-API will reject that if the two files have different datalayout or target triple, resulting in a runtime failure in QAT.

Reply:
@idavis @swernli  and @bettinaheim  the BC should be in the release artefacts IMO. The way I view it is that `ll` is equivalent to the "source" of the IR and bytecode is equivalent to the "binary" (even if there is a one-to-one correspondence). 

As such, I think it is questionable as to whether any IR should be checked in at all as they will always have platform dependence in form of the data layout and target machine. For instance, the IR will contain optimisation based on endianess assumptions that may not be true. Hence, the "right" thing to do would be to recompile the stdlib for the target platform (as we do for any other programming language).

Reply:
Regardless, this is a broader discussion and out of scope for this PR. We possibly need to understand what makes an IR target agnostic and how link-time optimisation works before making a decision on this. For this reason, I suggest that we create a ticket for this discussion. Are you OK with that  @bettinaheim?